### ISSUE TITLE
Mouse commands doesn't work

### ISSUE THREAD

**Issue by freitass** [majutsushi/tagbar#14]
Whenever I click on the tagbar window vim enters in visual mode and starts selecting everything on the window that was on focus before, from the original position of the cursor to the current mouse position (and the selection changes as if I was pressing the left mouse button).

Notes:
- NERDTree plugin is also installed and the mouse commands work just fine there.
- I'm using ` ` `mouse=a` ` `

**Comment by blochl** [comment]
Hi,
I can confirm this bug on Arch, Vim 8.0, rxvt-unicode.
Strangely, it began to appear only after some update (to Vim or Tagbar) cause a year ago it didn't happen.
NERDtree works fine for me in all the terminals, and Tagbar works fine as well in Gvim, or in Xfce terminal. The problem is just with Tagbar, just in rxvt-unicode.
Setting `ttymouse` to `urxvt` solves it on urxvt, and does not harm in other terminals, so that is what I'm using now.

**Comment by kettro** [comment]
I'm on cygwin, using vim 8.0, $TERM=xterm-256color. I encountered the problem when I changed my text size from 11pt to 10pt font. Returning to 11pt font fixed the issue.

**Comment by stephenafamo** [comment]
Changing my text size to >=11pt also fixed the issue for me.
Ubuntu 16.04 Vim 7.4

**Comment by feltech** [comment]
Same problem here with
* Ubuntu 17.10.
* QTerminal.
* `export TERM="xterm-256color"`.
* Tried all `ttymouse` settings within vim.
* Only thing that works for me is font 10pt or more.

**Comment by dpruessner** [comment]
Having same problem on

- OSX 10.14
- iTerm2 Build 3.1.5
- VIM - Vi IMproved 8.1 (2018 May 18, compiled Nov 16 2018 11:20:51) with mouse options:
- +mouse_netterm +mouse_sgr -mouse_sysmouse +mouse_urxvt +mouse_xterm +mouse +mouse_dec -mouse_gpm -mouse_jsbterm

Migrating from Taglist (which mouse left-click + scrolling works) to Tagbar and found that my mouse does not work on the Tagbar window

**Comment by open-trade** [comment]
> Having same problem on
> 
> * OSX 10.14
> * iTerm2 Build 3.1.5
> * VIM - Vi IMproved 8.1 (2018 May 18, compiled Nov 16 2018 11:20:51) with mouse options:
>   
>   * +mouse_netterm +mouse_sgr -mouse_sysmouse +mouse_urxvt +mouse_xterm +mouse +mouse_dec -mouse_gpm -mouse_jsbterm
> * tmux 2.8
> 
> Migrating from Taglist (which mouse left-click + scrolling works) to Tagbar and found that my mouse does not work on the Tagbar window.
> 
> Found that mouse does work in a straight terminal, but does not work in a _tmux_ session. Other plugins work with mouse properly in _tmux_ session, but not Tagbar.

I also have the same issue on jupyterlab terminal

**Comment by alerque** [comment]
I'm re-opening this issue because it is apparent from comments that something is still amiss. That being said I can't actually reproduce it myself. The only place I ever use vim _not_ inside `tmux` is [inside Firefox](https://github.com/glacambre/firenvim/). I don't have any trouble with the mouse in tagbar, but I'm  also using Neovim, so I suspect this is a vim configuration issue.

Can those of you who have this problem try to narrow down what the common factor is between you?

**Comment by open-trade** [comment]
> > Having same problem on
> > 
> > * OSX 10.14
> > * iTerm2 Build 3.1.5
> > * VIM - Vi IMproved 8.1 (2018 May 18, compiled Nov 16 2018 11:20:51) with mouse options:
> >   
> >   * +mouse_netterm +mouse_sgr -mouse_sysmouse +mouse_urxvt +mouse_xterm +mouse +mouse_dec -mouse_gpm -mouse_jsbterm
> > * tmux 2.8
> > 
> > Migrating from Taglist (which mouse left-click + scrolling works) to Tagbar and found that my mouse does not work on the Tagbar window.
> > Found that mouse does work in a straight terminal, but does not work in a _tmux_ session. Other plugins work with mouse properly in _tmux_ session, but not Tagbar.
> 
> I also have the same issue on jupyterlab terminal

latest nvim works fine

**Comment by dzhang-b** [comment]
Using `MINGW64` terminal simulator from `Git for windows`, When the the font size is 9, the mouse wont work inside tagbar. ( `Tmux 2.1` and `vim 8.1 Included patches: 1-2237`

Change the font size >= 12 seems work.

**Comment by alerque** [comment]
Thanks for helping narrow this down @dzhang-b. I would suggest that you are probably looking at a bug in your terminal emulation rather than in Tagbar though. If mouse signals working or not working change at different font sizes it sounds like the terminal itself is broken.

Out of curiosity you say the mouse works elsewhere in vim at the same time?

**Comment by dzhang-b** [comment]
@alerque Yes. Vim part have no problem (For both vim 7.4 and 8.1) to handle the mouse input.
Sometimes when I try to click the tagbar, whole vim freezes that I have to terminate the whole tmux panel. It is not a tmux issue. It has to do with vim, tagbar and terminal emulator interaction somehow.

**Comment by justeene** [comment]
win10 wsl debian:
tagbar when :mouse=a not work,but NERDTree can mouse click.

**Comment by alerque** [comment]
I'd be happy to help review and merge a PR that fixes whatever this elusive issue is, but I cannot reproduce it and I do not have anything even resembling some of the setups described (no Win-10→WSL→Debian for example) so I am unable to reproduce.

If somebody that actually has this problem figures out how to fix it, please feel free to open a PR. It would be welcomed.

**Comment by yaocz** [comment]
I have the same issue with ubuntu and archlinux,When using Konsole terminal open vim.If extended the terminal width,the issue will happen.When using Xterm or Terimte terminal open vim in tmux.It seems to happen.In other words,If issue happened,Reducing terminal width will work.

**Comment by alerque** [comment]
@yaocz Interesting information, thanks for reporting. Is there a specific terminal size in columns after which you see the bug? Is it only after resize or if the size is set before opening VIM too? Does it go away if you resize a window to be smaller after seeing the problem?

**Comment by yaocz** [comment]
@alerque Yes,I always opened the vim or vim in tmux with tagbar windows when resizing terminal width in real time.Indeed,the problem have a boundary of terminal size in columns.If i maximized terminal windows,It immediately happen in my bigger monitor.When i used another small monitor doesn't happen,Then i try to cancel maximize and intentionally extend width.As expected,It happened.
  Of course,Only g:tagbar_position='right' will happen the issue.And g:tagbar_width size is also affect that maximize windows doesn't happen the issue in my bigger monitor when the size is great than default size.

**Comment by Piping** [comment]
@yaocz  I can confirm the behavior and it only happens for `g:tagbar_position='right'` and has to do terminal column sizes.

**Comment by krshrimali** [comment]
I can reproduce this issue on my Pop OS 22.04 system. Thanks to their auto-tiling, it made sense that the issue is not just about terminal size, but also about the tag length. Here are my observations:

1. Write a huge function whose tag declaration does not fit the terminal size. (maybe just resize the terminal, or enter a split mode if you are on Mac/Linux - if you are using WSL, please use Windows Terminal and split the terminal).
2. That means when you click on the tag when the bar appears on the right, and if the length of the tag declaration exceeds the terminal size, and if it has to take >1 lines to be listed on the bottom -> then it waits for the user to hit enter in order to scroll further (up/down).

Now do the same thing, but with a file having all functions where declarations can be accommodated in the terminal size you have, this problem will not be there.

I'm thinking of a solution but will have to take a look at the source code. Please let me know if anyone faces any issues in reproducing this.

**Comment by urayoru113** [comment]
Same issue happening in `wsl window terminal`.
And I found only a small position can mouse clicking work.
I think tagbar got the **wrong** length of variable or function name.

**Comment by alerque** [comment]
That diagnosis sounds probable. Again I don't have access to Windows but we'd be happy to facilitate a PR if somebody tracks this down and contributes a fix for the benefit of other Windows users.
